File: src/test/java/de/fraunhofer/aisec/codyze/crymlin/ConstantValueTest.java

package de.fraunhofer.aisec.codyze.crymlin;

import de.fraunhofer.aisec.codyze.analysis.ConstantValue;
import org.junit.jupiter.api.Test;

import java.util.Optional;

import static org.junit.jupiter.api.Assertions.assertEquals;

class ConstantValueTest {

	@Test
	void testTryOfInteger() {
		int test = 42;
		Optional<ConstantValue> res = ConstantValue.tryOf(test);
		assertEquals(res.get().getValue(), test);
	}

	@Test
	void testTryOfFloat() {
		float test = 42f;
		Optional<ConstantValue> res = ConstantValue.tryOf(test);
		assertEquals(res.get().getValue(), test);
	}

	@Test
	void testTryOfBoolean() {
		Boolean test = true;
		Optional<ConstantValue> res = ConstantValue.tryOf(test);
		assertEquals(res.get().getValue(), test);
	}

	@Test
	void testTryOfString() {
		String test = "test";
		Optional<ConstantValue> res = ConstantValue.tryOf(test);
		assertEquals(res.get().getValue(), test);
	}
}

File: codyze_legacy/src/test/java/de/fraunhofer/aisec/codyze/crymlin/ConstantValueTest.kt
package de.fraunhofer.aisec.codyze.crymlin

import de.fraunhofer.aisec.codyze.analysis.resolution.ConstantValue
import kotlin.test.assertEquals
import org.junit.jupiter.api.Test

internal class ConstantValueTest {
    @Test
    fun testTryOfInteger() {
        val test = 42
        val res = ConstantValue.tryOf(test)
        assertEquals(res.get().value, test)
    }

    @Test
    fun testTryOfFloat() {
        val test = 42f
        val res = ConstantValue.tryOf(test)
        assertEquals(res.get().value, test)
    }

    @Test
    fun testTryOfBoolean() {
        val test = true
        val res = ConstantValue.tryOf(test)
        assertEquals(res.get().value, test)
    }

    @Test
    fun testTryOfString() {
        val test = "test"
        val res = ConstantValue.tryOf(test)
        assertEquals(res.get().value, test)
    }
}
